pubsub/azure: tests are failing with -record

```
google/go-cloud/pubsub/azurepubsub$ go test -v -record --run TestConformance/TestAs/asb
=== RUN   TestConformance
=== RUN   TestConformance/TestAs
=== RUN   TestConformance/TestAs/asb
--- FAIL: TestConformance (10.54s)
    --- FAIL: TestConformance/TestAs (10.54s)
        --- FAIL: TestConformance/TestAs/asb (10.54s)
            drivertest.go:505: failed to convert pubsub (code=Unknown): azurepubsub: no such subscription nonexistent-subscription (common.Retryable) to a common.Retryable
FAIL
exit status 1
FAIL	gocloud.dev/pubsub/azurepubsub	10.671s
```

--- a/pubsub/azurepubsub/azurepubsub.go
+++ b/pubsub/azurepubsub/azurepubsub.go
@@ -184,7 +184,7 @@ func openSubscription(ctx context.Context, sbNs *servicebus.Namespace, sbTop *se
 	}
 }
 
-// testSBSubscription ensures the subscripion exists before listening for incoming messages.
+// testSBSubscription ensures the subscription exists before listening for incoming messages.
 func (s *subscription) testSBSubscription(ctx context.Context) error {
 	if s.topicName == "" {
 		return errors.New("azurepubsub: driver.Subscription requires a Service Bus Topic")
@@ -201,10 +201,10 @@ func (s *subscription) testSBSubscription(ctx context.Context) error {
 		return err
 	}
 
-	// An empty SubscriptionEntity means no Service Bus Subscription exist for the given name.
+	// An empty SubscriptionEntity means no Service Bus Subscription exists for the given name.
 	se, _ := sm.Get(ctx, s.sbSub.Name)
 	if se == nil {
-		return fmt.Errorf("azurepubsub: no such subscription %v", s.sbSub.Name)
+		return fmt.Errorf("azurepubsub: no such subscription %q", s.sbSub.Name)
 	}
 	return nil
 }
@@ -312,8 +312,8 @@ func (s *subscription) SendAcks(ctx context.Context, ids []driver.AckID) error {
 		return err
 	}
 	defer client.Close()
-		
-	entityPath := s.topicName + "/Subscriptions/" + s.sbSub.Name		
+
+	entityPath := s.topicName + "/Subscriptions/" + s.sbSub.Name
 	audience := host + entityPath
 	err = cbs.NegotiateClaim(ctx, audience, client, s.sbNs.TokenProvider)
 	if err != nil {

--- a/pubsub/azurepubsub/azurepubsub_test.go
+++ b/pubsub/azurepubsub/azurepubsub_test.go
@@ -148,7 +148,7 @@ func TestConformance(t *testing.T) {
 type sbAsTest struct{}
 
 func (sbAsTest) Name() string {
-	return "asb"
+	return "azure"
 }
 
 func (sbAsTest) TopicCheck(top *pubsub.Topic) error {
@@ -178,16 +178,14 @@ func (sbAsTest) SubscriptionCheck(sub *pubsub.Subscription) error {
 func (sbAsTest) TopicErrorCheck(t *pubsub.Topic, err error) error {
 	var sbError common.Retryable
 	if !t.ErrorAs(err, &sbError) {
-		return fmt.Errorf("failed to convert %v (%T) to a common.Retryable", err, sbError)
+		return fmt.Errorf("failed to convert %v (%T) to a common.Retryable", err, err)
 	}
 	return nil
 }
 
 func (sbAsTest) SubscriptionErrorCheck(s *pubsub.Subscription, err error) error {
-	var sbError common.Retryable
-	if !s.ErrorAs(err, &sbError) {
-		return fmt.Errorf("failed to convert %v (%T) to a common.Retryable", err, sbError)
-	}
+	// We generate our own error for non-existent subscription, so there's no
+	// underlying Azure error type.
 	return nil
 }
 

